### ISSUE TITLE
Puzzles rewrite

### ISSUE THREAD

**Issue by scottbedard** [scottbedard/speedcube.site#80]
Our `<v-puzzle>` component is going to eventually need to be re-written. I'm opening this issue as a place to brainstorm features we'd like it to have.

- [x] All components should share the same renderer. This will improve performance, and make it so we can render multiple puzzles at once. [Click here for an example](https://threejs.org/examples/webgl_multiple_elements.html) ([source](https://github.com/mrdoob/three.js/blob/master/examples/webgl_multiple_elements.html)) of how to re-use renderers (replaces #51)
- [x] Rendering should be done in a `requestAnimationFrame` loop. This will let the browser optimize rendering and drop frames if necessary.
- ~~Be able to render any arbitrary frame in a replay. The idea here is that the replay component will simply transition the progress of the replay from `0` to `1`, and provide controls to step forward/backward through the solve.~~
- [ ] Update puzzle integration with solve page
- [ ] Update puzzle integration with replay page

> **Update:** I've decided not to let the parent component control the turning via props. Puzzles are being used very differently from page to page, so exposing this as props might cause lots of redundant logic. The `<v-puzzle>` component will continue to expose an api for the parent to work with.

**Comment by scottbedard** [comment]
This issue blocks #63

**Comment by scottbedard** [comment]
This is currently blocked by #35

**Comment by scottbedard** [comment]
Closing in favor if #92
